### PR TITLE
Don't show appointment sheet when patient is dead

### PR DESCRIPTION
**Pull request by msasikanth**
https://app.shortcut.com/simpledotorg/story/5098/go-back-to-the-previous-screen-when-done-back-is-clicked-in-the-patient-summary-screen-when-the-patient-is-marked-as-dead

### CODE REVIEW

File: app/src/main/java/org/simple/clinic/summary/PatientSummaryUpdate.kt [hunk 1]
```diff
@@ -54,6 +54,10 @@ class PatientSummaryUpdate : Update<PatientSummaryModel, PatientSummaryEvent, Pa
     }
   }
 
+  private fun backClicked(): Next<PatientSummaryModel, PatientSummaryEffect> {
+    return dispatch(GoBackToPreviousScreen)
```
Comment (vinaysshenoy) [inline app/src/main/java/org/simple/clinic/summary/PatientSummaryUpdate.kt]: The commit message includes the qualifier of the patient being dead, but it doesn't seem like we are checking the patient status in this commit. Should we reword the message?

File: app/src/main/java/org/simple/clinic/summary/PatientSummaryUpdate.kt [hunk 2]
```diff
@@ -54,6 +54,10 @@ class PatientSummaryUpdate : Update<PatientSummaryModel, PatientSummaryEvent, Pa
     }
   }
 
+  private fun doneClicked(): Next<PatientSummaryModel, PatientSummaryEffect> {
+    return dispatch(GoToHomeScreen)
```
Comment (vinaysshenoy) [inline app/src/main/java/org/simple/clinic/summary/PatientSummaryUpdate.kt]: Same comment as earlier about the commit message not matching the actual change in this commit.

File: app/src/test/java/org/simple/clinic/summary/PatientSummaryUpdateTest.kt [hunk 3]
```diff
@@ -995,6 +995,21 @@ class PatientSummaryUpdateTest {
...
+
+    updateSpec
+        .given(model)
+        .whenEvent(PatientSummaryDoneClicked(patientUuid))
+        .then(assertThatNext(
+            hasNoModel(),
+            hasEffects(LoadDataForDoneClick(patientUuid))
+        ))
+  }
+
```
Comment (vinaysshenoy) [inline app/src/test/java/org/simple/clinic/summary/PatientSummaryUpdateTest.kt:1012]: Given that we have added some Mobius style tests for existing user flows in `PatientSummaryScreen`, should we add the remaining effect handler tests and then remove the tests pertaining to this flow from the older regression tests?

**Review by vinaysshenoy — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/simple/clinic/summary/PatientSummaryUpdate.kt (hunk 1 above)
Comment (msasikanth) [reply app/src/main/java/org/simple/clinic/summary/PatientSummaryUpdate.kt]: Ah, I forgot I can just return `noChange`.  Making that change.

File: app/src/test/java/org/simple/clinic/summary/PatientSummaryUpdateTest.kt (hunk 3 above)
Comment (msasikanth) [reply app/src/test/java/org/simple/clinic/summary/PatientSummaryUpdateTest.kt:1012]: Yeah, I can do that in a different PR?

Comment (vinaysshenoy) [reply app/src/test/java/org/simple/clinic/summary/PatientSummaryUpdateTest.kt:1012]: Yup, that works.

**Review by vinaysshenoy — APPROVED** [review]

**Pull request merged**